src/codegen.cpp

```diff
@@ -8450,15 +8450,19 @@ static jl_llvm_functions_t
         cursor = -1;
     };
 
+    // If a pkgimage or sysimage is being generated, disable tracking.
+    // This means sysimage build or pkgimage precompilation workloads aren't tracked.
     auto do_coverage = [&] (bool in_user_code, bool is_tracked) {
-        return (coverage_mode == JL_LOG_ALL ||
+        return (jl_generating_output() == 0 &&
+                (coverage_mode == JL_LOG_ALL ||
                 (in_user_code && coverage_mode == JL_LOG_USER) ||
-                (is_tracked && coverage_mode == JL_LOG_PATH));
+                (is_tracked && coverage_mode == JL_LOG_PATH)));
     };
     auto do_malloc_log = [&] (bool in_user_code, bool is_tracked) {
-        return (malloc_log_mode == JL_LOG_ALL ||
+        return (jl_generating_output() == 0 &&
+                (malloc_log_mode == JL_LOG_ALL ||
                 (in_user_code && malloc_log_mode == JL_LOG_USER) ||
-                (is_tracked && malloc_log_mode == JL_LOG_PATH));
+                (is_tracked && malloc_log_mode == JL_LOG_PATH)));
```

Comment:
This is to avoid baking tracking-enabled native code into pkgimages, which is likely undesirable (though arguably reasonable in CI environments)

Reply:
Perhaps it could be allowed if `--check-bounds=yes` is also forced, as it is by Pkg.test, given that is unlikely in non-test sessions.

Reply:
Maybe @keno has an idea on how to handle `--check-bounds=yes`.

IIUC the problem is that `--check-bounds` doesn't just change the LLVM code, but actually the behaviour of inference and co. See https://github.com/JuliaLang/julia/pull/50239#issuecomment-1656872289

So right now I don't see away around us needing at least those two copies.